parlai/mturk/core/mturk_manager.py

```diff
@@ -53,6 +54,28 @@
 parent_dir = os.path.dirname(os.path.abspath(__file__))
 
 
+class LockFile():
+    flags = os.O_CREAT | os.O_EXCL | os.O_WRONLY
+
+    def __init__(self, filename):
+        self.filename = filename
+        self.fd = None
+
+    def __enter__(self):
+        while self.fd is None:
+            try:
+                self.fd = os.open(self.filename, self.flags)
+            except OSError as e:
+                if e.errno == errno.EEXIST:  # Failed as the file exists.
+                    pass
+            time.sleep(shared_utils.THREAD_SHORT_SLEEP)
```

Comment:
what does the sleep do here? just curious

Reply:
Just here to prevent a tight loop if a thread is waiting on the lock.